packages/frontend/src/components/wallet-migration/Disable2FA.jsx

```diff
@@ -0,0 +1,166 @@
+        font-size: 12px;
+        padding-top: 72px;
+        padding-bottom: 6px;
+    }
+
+    .title{
+        font-weight: 800;
+        font-size: 20px;
+        margin-top: 40px;
+    }
+`;
+
+const ButtonsContainer = styled.div`
+    text-align: center;
+    width: 100% !important;
+    display: flex;
+`;
+
+const StyledButton = styled(FormButton)`
+    width: calc((100% - 16px) / 2);
+    margin: 48px 0 0 !important;
+
+    &:last-child{
+        margin-left: 16px !important;
+    }
+`;
+
+
+const Disable2FAModal = ({ handleSetActiveView, onClose }) => {
+    const [state, localDispatch] = useImmerReducer(sequentialAccountImportReducer, {
+        accounts: []
+    });
+    const [loadingMultisigAccounts, setLoadingMultisigAccounts] = useState(true);
+    const initialAccountIdOnStart = useSelector(selectAccountId);
+    const initialAccountId = useRef(initialAccountIdOnStart);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const update2faAccounts = async () => {
+            const accounts = await wallet.keyStore.getAccounts(NETWORK_ID);
+            const getAccountWithAccessKeysAndType = async (accountId) => {
+                const keyType = await wallet.getAccountKeyType(accountId);
+                return { accountId, keyType };
+            };
+            const accountsKeyTypes = await Promise.all(
+                accounts.map(getAccountWithAccessKeysAndType)
+            );
+            localDispatch({
+                type: ACTIONS.ADD_ACCOUNTS,
+                accounts: accountsKeyTypes.reduce(((acc, { accountId, keyType }) => keyType === WalletClass.KEY_TYPES.MULTISIG ? acc.concat({ accountId, status: null }) : acc), [])
+            });
+            setLoadingMultisigAccounts(false);
+        };
+        setLoadingMultisigAccounts(true);
+        update2faAccounts();
+    }, []);
+
+    const failed = useMemo(() => state.accounts.some((account) => account.status === IMPORT_STATUS.FAILED), [state.accounts]);
+    const currentAccount = useMemo(() => !failed && state.accounts.find((account) => account.status === IMPORT_STATUS.PENDING), [failed, state.accounts]);
+    const batchDisableNotStarted = useMemo(() => state.accounts.every((account) => account.status === null), [state.accounts]);
+    const completedWithSuccess = useMemo(() => !loadingMultisigAccounts && state.accounts.every((account) => account.status === IMPORT_STATUS.SUCCESS), [state.accounts, loadingMultisigAccounts]);
+
+    useEffect(() => {
+        if (batchDisableNotStarted) {
+            initialAccountId.current = initialAccountIdOnStart;
+        }
+    },[initialAccountIdOnStart, batchDisableNotStarted]);
+
+    useEffect(() => {
+        const disable2faForCurrentAccount = async () => {
+            try {
+                await dispatch(switchAccount({accountId: currentAccount.accountId}));
+                const account = await wallet.getAccount(currentAccount.accountId);
+                await account.disableMultisig();
```

Comment:
We have `disableMultisig` redux action, probably we should use it here and additionally dispatch `fetchRecoveryMethods` as well to refresh the redux state.

Reply:
Agreed 👍 . Initially I was trying to disable 2fa without switching accounts in the wallet but was not able to. Since we are now switching before disable we should use the redux action